[DataGrid] Keyboard arrow should scroll the body

yaredtsy opened the pull request:
Fixes #4282

**#Problem**

when using the `arrow` or clicking on the cell that is out of the viewport from the browser. it will not scroll the browser body.

![160253962-d9117972-7804-4b86-8da1-2ed37d27b445](https://user-images.githubusercontent.com/34584266/205311327-e53d70e8-c9a4-4a13-834e-3aac46a173d8.gif)

**#Soultion**

check if the focused cell is out of the viewport. and if it is out of the viewport call `scrollIntoView` and scroll to the nearest position that will make it visible in the viewport.

oliviertassinari reviewed (commented):
Could you rebase on `next`. We are focusing on this branch, thanks

yaredtsy commented:
> This solution bypasses the focus state. The focus state could point to cell A, but cell B is clicked and it will scroll to cell B.

fixed, I moved it to the `useEffect` which handles the focus.
